apis/management.cattle.io/v3public/authn_types.go

```diff
@@ -37,11 +37,19 @@ type GithubProvider struct {
 	RedirectURL string `json:"redirectUrl"`
 }
 
-type GithubLogin struct {
+type CodeBasedLogin struct {
```

Comment:
We are changing the Login Action Input, I guess this needs a UI change for Github provider? Do we have any integrations(automation test etc) other than UI that does github auth

Reply:
Also I guess we need similar change for Azure AD provider?

Reply:
Azure AD Login has it's own separate types schema https://github.com/rancher/types/blob/master/apis/management.cattle.io/v3public/authn_types.go#L69

Yeah maybe UI change will be required. I don't think we have any automation tests for Github. I think creating separate schema for google oauth login makes sense then like Azure AD's?

Reply:
Yeah if our aim it to consolidate all "code" based logins, then we should also change AzureAD. But since it is already a v3public API, I guess its better to add new rather than affect existing auth providers